cmd/migration-tool/main.go

```diff
@@ -0,0 +1,524 @@
+			APIVersion: "release.openshift.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      release.Name,
+			Namespace: release.SourceImageStreamNamespace,
+			Labels: map[string]string{
+				releasecontroller.ReleasePayloadLabelImageStream:    config.To,
+				releasecontroller.ReleasePayloadLabelImageStreamTag: release.Name,
+			},
+			Annotations: map[string]string{},
+		},
+		Spec: v1alpha1.ReleasePayloadSpec{
+			PayloadCoordinates: v1alpha1.PayloadCoordinates{
+				Namespace:          release.SourceImageStreamNamespace,
+				ImagestreamName:    config.To,
+				ImagestreamTagName: release.Name,
+			},
+		},
+		Status: v1alpha1.ReleasePayloadStatus{
+			BlockingJobResults:  []v1alpha1.JobStatus{},
+			InformingJobResults: []v1alpha1.JobStatus{},
+			AnalysisJobResults:  []v1alpha1.JobStatus{},
+		},
+	}
+
+	for alias, job := range config.Verify {
+		if job.Disabled {
+			continue
+		}
+		for k, v := range release.Status {
+			var status *v1alpha1.JobStatus
+			if k != alias {
+				continue
+			}
+
+			jobName := job.ProwJob.Name
+
+			if job.AggregatedProwJob != nil {
+				jobName = fmt.Sprintf("%s-%s", alias, defaultAggregateProwJobName)
+			}
+
+			result := v1alpha1.JobRunResult{
+				Coordinates: v1alpha1.JobRunCoordinates{
+					Name:      jobName,
+					Namespace: prowNamespace,
+				},
+				State:               v1alpha1.JobRunState(v.State),
+				HumanProwResultsURL: v.URL,
+			}
+
+			status = payload.Status.FindJob(job.ProwJob.Name, alias, job.Optional, job.MaxRetries)
+			if status == nil {
+				newJobStatus := v1alpha1.JobStatus{
+					JobAlias:   alias,
+					JobName:    job.ProwJob.Name,
+					MaxRetries: job.MaxRetries,
+					Optional:   job.Optional,
+					JobRunResults: []v1alpha1.JobRunResult{
+						result,
+					},
+				}
+				switch {
+				case job.AggregatedProwJob != nil:
+					// The Aggregator job can be blocking or informing
+					if job.Optional {
+						payload.Status.InformingJobResults = append(payload.Status.InformingJobResults, newJobStatus)
+					} else {
+						payload.Status.BlockingJobResults = append(payload.Status.BlockingJobResults, newJobStatus)
+					}
+					// The Analysis jobs have their own category
+					analysisJobStatus := newJobStatus.DeepCopy()
+					analysisJobStatus.JobName = job.ProwJob.Name
+					analysisJobStatus.JobAlias = alias
+					analysisJobStatus.JobRunResults = []v1alpha1.JobRunResult{}
+					payload.Status.AnalysisJobResults = append(payload.Status.AnalysisJobResults, *analysisJobStatus)
+				case job.Optional:
+					payload.Status.InformingJobResults = append(payload.Status.InformingJobResults, newJobStatus)
+				case !job.Optional:
+					payload.Status.BlockingJobResults = append(payload.Status.BlockingJobResults, newJobStatus)
+				default:
+					klog.Warningf("Unable to classify job type: [%s] %s", config.Name, alias)
+				}
+			} else {
+				status.JobRunResults = append(status.JobRunResults, result)
+			}
+		}
+	}
+
+	condition := v1alpha1.ReleasePayloadCondition{
```

Comment:
calculate this separately from calculating your job results.  Your conditions will logically be calculated from different inputs.